core/include/vecmem/containers/impl/vector.ipp

```diff
@@ -13,15 +13,21 @@ namespace vecmem {
    data::vector_view< TYPE >
    get_data( std::vector< TYPE, ALLOC >& vec ) {
 
-      return { vec.size(), vec.data() };
+      return {
+         static_cast<
+            typename data::vector_view< TYPE >::size_type >( vec.size() ),
+         vec.data() };
    }
 
    template< typename TYPE, typename ALLOC >
    VECMEM_HOST
    data::vector_view< const TYPE >
    get_data( const std::vector< TYPE, ALLOC >& vec ) {
 
-      return { vec.size(), vec.data() };
+      return {
+         static_cast<
+            typename data::vector_view< const TYPE >::size_type >( vec.size() ),
```

Comment:
This is because the size_types can be different?

Reply:
Yes. :smile: With this PR I change `vecmem::data::vector_view<T>::size_type` and `vecmem::device_vector<T>::size_type` to `unsigned int`. (From the previous `std::size_t`.) The conversion from `std::size_t` to `unsigned int` happens in these functions "canonically"... (And the compiler would give us warnings about loss of precision if we didn't use an explicit cast.)

Reply:
Ok I see.